### PR TITLE
TF-2679 Implement server settings repository

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2615 
- #2679

### What's in this PR

- Implementation of Server settings repository

### CODE REVIEW

File: test/features/server_settings/data/repository/server_settings_repository_impl_test.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        // act
+        final result = await serverSettingsRepository
+          .updateServerSettings(accountId, serverSettings);
+
+        // assert
+        expect(result, serverSettings);
+      });
+
+      test('should rethrow exception when ServerSettingsDataSource throws exception',
+      () async {
```
Comment (dab246) [inline test/features/server_settings/data/repository/server_settings_repository_impl_test.dart:66]:
```suggestion
      test('should rethrow exception when ServerSettingsDataSource throws exception', () async {
```

File: test/features/server_settings/data/repository/server_settings_repository_impl_test.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  final accountId = AccountId(Id('123'));
+  final serverSettingsDataSource = MockServerSettingsDataSource();
+  final serverSettingsRepository = ServerSettingsRepositoryImpl(
+    serverSettingsDataSource,
+  );
+
+  group('server settings repository', () {
+    group('get server settings', () {
+      test('should return value when ServerSettingsDataSource returns value',
+      () async {
```
Comment (dab246) [inline test/features/server_settings/data/repository/server_settings_repository_impl_test.dart:25]:
```suggestion
      test('should return value when ServerSettingsDataSource returns value', () async {
```

File: test/features/server_settings/data/repository/server_settings_repository_impl_test.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        // assert
+        expect(
+          () => serverSettingsRepository.getServerSettings(accountId),
+          throwsA(const TypeMatcher<NotFoundServerSettingsException>()));
+      });
+    });
+
+    group('update server settings', () {
+      test('should return value when ServerSettingsDataSource returns value',
+      () async {
```
Comment (dab246) [inline test/features/server_settings/data/repository/server_settings_repository_impl_test.dart:52]: idem

File: test/features/server_settings/data/repository/server_settings_repository_impl_test.dart (hunk 1 above)
Comment (tddang-linagora) [reply test/features/server_settings/data/repository/server_settings_repository_impl_test.dart]: done

File: test/features/server_settings/data/repository/server_settings_repository_impl_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/features/server_settings/data/repository/server_settings_repository_impl_test.dart]: done

File: test/features/server_settings/data/repository/server_settings_repository_impl_test.dart (hunk 3 above)
Comment (tddang-linagora) [reply test/features/server_settings/data/repository/server_settings_repository_impl_test.dart]: done

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
please rebase

**Review by hoangdat — APPROVED** [review]

**Pull request merged**